File: candidate_graph/pre_data/extract.js
var fs = require('fs')
module.exports.extract = function (arr_address){
    for(var i = 0; i < arr_address.length;i++){
         var name_city = arr_address[i]["name"] //city
         var city = arr_address[i]
             var district = city["district"] //district
             for(var k = 0; k < district.length; k++){
                 var name_dis = district[k]["name"]
                 var street = district[k]["street"] //street
                 for( var m = 0; m < street.length;m++){
                        var str = {
                            "street" : street[m]["name"],
                            "district" : name_dis,
                            "city" : name_city,
                            "code" : arr_address[i]["code"],
                            "country" : "Việt Nam"
                        }
                        fs.appendFileSync('./pre_data.json', JSON.stringify(str))
                        fs.appendFileSync('./pre_data.json',"\n")
                 }
                 var ward = district[k]["ward"] //ward
                 for( var n = 0; n < ward.length;n++){
                        var war = {
                            "ward" : ward[n]["name"],
                            "district" : name_dis,
                            "city" : name_city,
                            "code" : arr_address[i]["code"],
                            "country" : "Việt Nam"
                        }
                        fs.appendFileSync('./pre_data.json', JSON.stringify(war))
                        fs.appendFileSync('./pre_data.json',"\n")
                 }
                 var project = district[k]["project"] //project
                 for( var p = 0; p < project.length;p++){
                        var pro = {
                            "project" : project[p]["name"],
                            "district" : name_dis,
                            "city" : name_city,
                            "lat" : project[p]["lat"],
                            "lng" : project[p]["lng"],
                            "code" : arr_address[i]["code"],
                            "country" : "Việt Nam"
                        }
                        fs.appendFileSync('./pre_data.json', JSON.stringify(pro))
                        fs.appendFileSync('./pre_data.json',"\n")
                 }
             }
        }
}

File: candidate_graph/pre_data/extract.py
import json
import unicodedata as ud


def normalize(str):
    return ud.normalize('NFC', str)

def extractAddress(arr_adress):
    with open('./pre_data.json','a') as f:
        for i in range(len(arr_adress)):
            name_city = normalize(arr_adress[i]["name"].lower()) #city
            district = arr_adress[i]["district"]
            for j in range(len(district)):
                name_dis = normalize(district[j]["name"].lower()) #district
                street = district[j]["street"] #street
                for k in range(len(street)):
                    str = {
                        "street": normalize(street[k]["name"].lower()),
                        "district": name_dis,
                        "city": name_city,
                        "code": arr_adress[i]["code"],
                        "country": "việt nam"
                    }
                    json.dump(str, f, ensure_ascii=False,separators=(',',':'))
                    f.write("\n")
                ward = district[j]["ward"] #ward
                for n in range(len(ward)):
                    war = {
                        "ward": normalize(ward[n]["name"].lower()),
                        "district": name_dis,
                        "city": name_city,
                        "code": arr_adress[i]["code"],
                        "country": "việt nam"
                    }
                    json.dump(war, f, ensure_ascii=False,separators=(',',':'))
                    f.write("\n")

                project = district[j]["project"] #project
                for m in range(len(project)):
                    pro = {
                        "project": normalize(project[m]["name"].lower()),
                        "district": name_dis,
                        "city": name_city,
                        "code": arr_adress[i]["code"],
                        "country": "việt nam",
                        "lat": project[m]["lat"],
                        "lng": project[m]["lng"]
                    }
                    json.dump(pro, f, ensure_ascii=False,separators=(',',':'))
                    f.write("\n")
